src/sweeper/__main__.py

```diff
@@ -4,60 +4,105 @@
 from .sweepers.duplicates import DuplicateTest
 from .sweepers.empties import EmptyTest
 
 
 def main():
-    '''Main entry point for program. Parse arguments and pass to sweeper modules
+    '''Main entry point for program. Parse arguments and pass to sweeper modules.
     '''
-
     args = docopt(__doc__, version='1.0.0')
 
     #: backup input file before quality checks
     if args['--backup-to']:
-        backup.backup_data(os.path.join(args['--workspace'], args['<table_name>']), args['--backup-to'])
+        backup.backup_data(os.path.join(args['--workspace'], args['--table-name']), args['--backup-to'])
 
+    #: check what quality check to run.
     if args['duplicates']:
-        #: do something
-        print(args)
-        # duplicates.find(args['--input'])
-        sweeper = DuplicateTest(args['--workspace'], args['<table_name>'])
+        call_sweeper(args, "DuplicateTest")
     elif args['invalids']:
-        #: do something
-        print(args)
+        call_sweeper(args, "InvalidsTest")
     elif args['empties']:
-        print(args)
+        call_sweeper(args, "EmptyTest")
+    elif args['all-checks']:
+        call_sweeper(args, "DuplicateTest")
+        call_sweeper(args, "EmptyTest")
+
+
+def call_sweeper(args, sweeper_test):
+    '''
+    orchestrate the sweeper calls.
 
-        sweeper = EmptyTest(args['--workspace'], args['<table_name>'])
+    args: the docopt arguments.
+    sweeper_test: the name of the sweeper function to call.
+    '''
+    print(args)
+
+    #: if table name was provided, then run quality check on single table.
+    if args['--table-name']:
+        #: call the quality check.
+        if sweeper_test == "DuplicateTest":
+            sweeper = DuplicateTest(args['--workspace'], args['--table-name'])
+            save_report(args, sweeper, args['--table-name'], 'duplicates')
+        elif sweeper_test == "EmptyTest":
+            sweeper = EmptyTest(args['--workspace'], args['--table-name']) 
+            save_report(args, sweeper, args['--table-name'], 'empties')
+        #: attempt to fix flagged items.
+        if args['--try-fix']:
+            sweeper.try_fix()
+    #: if table name was not provided, then run quality check on all tables in the workpace.
+    else:
+        #: get a list of feature classes that are contained in the workspace.
+        featureclasses = workspace_info.get_featureclasses(args['--workspace'])
+        for fc in featureclasses:
+            print(fc)
+            #: call the quality check.
+            if sweeper_test == "DuplicateTest":
+                sweeper = DuplicateTest(args['--workspace'], fc)
+                save_report(args, sweeper, fc, 'duplicates')
+            elif sweeper_test == "EmptyTest":
+                sweeper = EmptyTest(args['--workspace'], fc)
+                save_report(args, sweeper, fc, 'empties')
+            #: attempt to fix flagged items.
+            if args['--try-fix']:
+                sweeper.try_fix()
```

Comment:
i'm not sure this extra method benefits the architecture much in its current state. It is more moving code without simplifying things. We should probably rework this section.

Reply:
yeah, we creatinly can.  i was stuck on how to best do that.

Reply:
how do you want to sort this out? Pair programming, a discussion and then you go off and code or what do you think?

Reply:
i'm okay if you want to code something out.  is that 'pair programming'

Reply:
no that's called you do it :) Pair programming would be you and I screen sharing and writing code together

Reply:
I can code/stub something out, lets get all the other stuff sorted out first since it sounds like the team had a change of heart

Reply:
the new refactor looks much nicer so far.